src/GeometryInstancing/Program.cs

```diff
@@ -0,0 +1,261 @@
+                default: throw new System.InvalidOperationException();
+            }
+
+            string entryPoint = stage == ShaderStages.Vertex ? "VS" : "FS";
+            string path = Path.Combine(System.AppContext.BaseDirectory, "Shaders", $"{stage.ToString()}.{extension}");
+            byte[] shaderBytes = File.ReadAllBytes(path);
+            return _graphicsDevice.ResourceFactory.CreateShader(new ShaderDescription(stage, shaderBytes, entryPoint));
+        }
+
+        private static void Draw()
+        {
+            // Begin() must be called before commands can be issued.
+            _commandList.Begin();
+
+            // We want to render directly to the output window.
+            _commandList.SetFramebuffer(_graphicsDevice.SwapchainFramebuffer);
+            _commandList.SetFullViewports();
+            _commandList.ClearColorTarget(0, RgbaFloat.Black);
+
+            _commandList.UpdateBuffer(_cameraProjViewBuffer,0,_camera.ViewMatrix);
+            _commandList.UpdateBuffer(_cameraProjViewBuffer,64,_camera.ProjectionMatrix);
+
+            // Set all relevant state to draw our quad.
+            _commandList.SetVertexBuffer(0,_vertexBuffer);
+            _commandList.SetIndexBuffer(_indexBuffer,IndexFormat.UInt16);
+            _commandList.SetVertexBuffer(1,_xOffsetBuffer);
+            _commandList.SetPipeline(_pipeline);
+            // Set uniforms
+            _commandList.SetGraphicsResourceSet(0,_resourceSet); // Always after SetPipeline
+            // Issue a Draw command for two instances with 4 indices.
+            _commandList.DrawIndexed(
+                indexCount: 4,
+                instanceCount: 2,
+                indexStart: 0,
+                vertexOffset: 0,
+                instanceStart: 0);
+
+            // End() must be called before commands can be submitted for execution.
+            _commandList.End();
+            _graphicsDevice.SubmitCommands(_commandList);
+
+            // Once commands have been submitted, the rendered image can be presented to the application window.
+            _graphicsDevice.SwapBuffers();
+        }
+
+        private static void DisposeResources()
+        {
+            _pipeline.Dispose();
+            _vertexShader.Dispose();
+            _fragmentShader.Dispose();
+            _commandList.Dispose();
+            _vertexBuffer.Dispose();
+            _indexBuffer.Dispose();
+            _graphicsDevice.Dispose();
+            _resourceSet.Dispose();
+            _resourceLayout.Dispose();
+            _xOffsetBuffer.Dispose();
+            _cameraProjViewBuffer.Dispose();
+        }
+    }
+
+    struct VertexPositionColor
+    {
+        public const uint SizeInBytes = 24;
+        public Vector2 Position;
+        public RgbaFloat Color;
+        public VertexPositionColor(Vector2 position, RgbaFloat color)
+        {
+            Position = position;
+            Color = color;
+        }
+    }
+
+    public struct VertexPositionColour

```

Comment:
Nit: use `Color` instead of `Colour` 😄

Reply:
Haha. yes I adopted uk english for my project. But I seem to be the only one xD. Ill change it to US